hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/BackendEntryIterator.java

```diff
@@ -115,8 +118,17 @@ protected final boolean reachLimit() {
     }
 
     protected final boolean reachLimit(long count) {
-        checkInterrupted();
-        return this.query.reachLimit(count);
+        try {
+            checkInterrupted();
+            return this.query.reachLimit(count);
+        } catch (BackendException e) {
+            try {
+                this.close();
+            } catch (Exception ex) {
+                LOG.warn("Failed to close backend entry iterator for interrupted query", ex);
+            }
+            throw e;
+        }
```

Comment:
also catch `QueryResults.toList(Iterator<T> iterator)` which may cause leak if exception occurred, and close in finally block:
```java
try {
    return new ListIterator<>(Query.DEFAULT_CAPACITY, iterator);
} finally {
    CloseableIterator.closeIterator(iterator);
}
```